packages/babel-plugin-proposal-explicit-resource-management/src/index.ts

```diff
@@ -160,6 +160,60 @@ export default declare(api => {
         }
       }
     },
+    SwitchStatement(path, state) {
+      let ctx: t.Identifier | null = null;
+      let needsAwait = false;
+
+      const { cases } = path.node;
+      for (const c of cases) {
+        for (const stmt of c.consequent) {
+          if (isUsingDeclaration(stmt)) {
+            ctx ??= path.scope.generateUidIdentifier("usingCtx");
+
+            const isAwaitUsing = stmt.kind === "await using";
+            needsAwait ||= isAwaitUsing;
+
+            stmt.kind = "const";
+            for (const decl of stmt.declarations) {
+              decl.init = t.callExpression(
+                t.memberExpression(
+                  t.cloneNode(ctx),
+                  isAwaitUsing ? t.identifier("a") : t.identifier("u"),
+                ),
+                [decl.init],
+              );
+            }
+          }
+        }
+      }
+      if (!ctx) return;
+
+      if (!process.env.BABEL_8_BREAKING && !state.availableHelper("usingCtx")) {
+        throw path.buildCodeFrameError(
+          "We cannot transpile this, please update `@babel/core`.",
+        );
```

Comment:
Could you throw this on the first `using` declaration instead of the switch statement?

Saying something "`using` declarations inside `switch` statements are not supported by your current `@babel/core` version, please update to a more recent one"